### DISCUSSION TITLE
Cant get it to generate or find models

### DISCUSSION THREAD

**Discussion by ryan13mt** [painebenjamin/app.enfugue.ai#2] [Q&A] [answered]
Not sure i'm doing anything wrong. I downloaded the windows files, extracted, opened the exe and managed to open the page. I clicked to download the base models shown on the first screen. Downloaded some extra models from civitai screen inside the app.

My top left model chooser is empty, tried restarting app and also restarting machine, same thing.

When clicking enfugue, it initializes and then stops. I can see the GPU memory usage go from 2gb to 8gb which is all i have and it stays like this even after failing to generate so memory is not being released it seems. 

Anyone else have such problems? I have auto1111 set up and working ok so this is not my first try at SD and my pc can handle it.

**Comment by painebenjamin** [comment]
Hi Ryan,

There's definitely some room for me to explain better what you need to do with models. I tried to update the README last night with more information about this, but models aren't just a singular checkpoint, you need to configure models that reference the downloaded checkpoint and other model(s).

![image](https://github.com/painebenjamin/app.enfugue.ai/assets/57536852/53ef0cc2-ae34-4f43-a688-860033b024f0)

For example, after making this model here, The model picker allows me to select "Realism" and is using the cyberrealistic checkpoint.

The reason for this style of configuration over direct file picking is primarily for the maintenance of TensorRT engines. Engines are specific to the checkpoint, size, and other weights (LoRA, TI) that are in it at compile time, so if the user changes any of those variables, the engine will no longer work.

After you initially clicked "Enfugue," about how long did you wait before it looked like nothing was going to happen? The first invocation can take a bit longer than subsequent ones, as it reaches out to Huggingface to pull other details of the models.

**Reply by ryan13mt** [reply]
Not sure what exactly it's doing cause i got to your explanation of how to create a model but get the cude out of memory error around 1 minute after pressing enfugue. But i'm noticing after failing there is still a process taking up all my 8GB of VRAM. And to get that ram back i have to manually end on of the Enfugue processes in the task manager.

Is there a way we can look at the applications logs for some debugging?

And what do i need to delete for it to think its a fresh install?

Thanks! love the work!

**Reply by painebenjamin** [reply]
Yes please! Logs are at `~/.cache/enfugue.log`, it would be excellent if you could post them.

I'm happy to hear you at least got *some* response - an out-of-memory error is at least not a "I have no idea what I'm doing" error.

With respect to the processes - I'm sorry they didn't clean themselves up. In theory there are two processes, the main server and the diffusion engine. The engine process stays alive in use, but will eventually exit itself to free up VRAM when Enfugue is not in use - it should take less than 30 seconds. If you're willing and able, next time you exit Enfugue, please give it a short moment and see if it cleans up after itself.

Also, I really appreciate people who say "love the work" even when it doesn't do anything for you. Thanks for understanding how hard software development can be sometimes :)

**Reply by ryan13mt** [reply]
I'm a dev myself so i know what it's like haha. Here is the log file, i emptied it just before pressing enfugue. 
[enfugue.log](https://github.com/painebenjamin/app.enfugue.ai/files/11895384/enfugue.log)

After getting the error in the UI it still shows full VRAM usage even after the failure. 

![image](https://github.com/painebenjamin/app.enfugue.ai/assets/6053618/363071d0-424c-4e6c-abb1-b6a2db42ba9b)



The only thing i could find in it is the CUDA error, not sure what else there is in the logs except that repeating one every few seconds.

**Reply by painebenjamin** [reply]
Brilliant, thank you for the log. It appears to be _just_ failing on your VRAM amount. The initialization process that it's trying to do shouldn't be taking _that_ much VRAM, though. Can you let me know what model(s) you're using (including precision and full/pruned,) if you're inpainting (I can't see the whole canvas, I'm guessing you aren't), or if you're trying TensorRT (I don't think you are, but making sure) so I can try and recreate the exact scenario?

Thanks again!

**Reply by ryan13mt** [reply]
Im using the one you download at the start, the sd 1.5. 

![image](https://github.com/painebenjamin/app.enfugue.ai/assets/6053618/0303ce7f-adec-4200-9ea7-18f750909b51)
![image](https://github.com/painebenjamin/app.enfugue.ai/assets/6053618/73603736-927e-4a02-bd64-dad2267fe016)

And these are the parameters i can see in the history tab:

```
{
   "model":"C:\\Users\\User\\.cache\\enfugue\\checkpoint\\sd-v1-5-inpainting.ckpt",
   "lora":[
      
   ],
   "inversion":[
      
   ],
   "width":512,
   "height":512,
   "size":512,
   "seed":null,
   "prompt":"orc character",
   "negative_prompt":"",
   "image":null,
   "image_callback_steps":10,
   "nodes":[
      {
         "width":512,
         "height":512,
         "prompt":"orc character",
         "negative_prompt":"",
         "controlnet":null,
         "conditioning_scale":1,
         "strength":0.8,
         "num_inference_steps":50,
         "guidance_scale":7.5,
         "remove_background":false,
         "process_control_image":true,
         "scale_to_model_size":true,
         "image":null,
         "mask":null,
         "control_image":null,
         "children":[
            
         ],
         "bounds":[
            [
               0,
               0
            ],
            [
               512,
               512
            ]
         ]
      }
   ],
   "samples":1,
   "upscale":{
      "method":[
         "esrgan"
      ],
      "amount":1,
      "iterative":false,
      "diffusion":false
   },
   "chunking_size":64,
   "chunking_blur":64,
   "build_tensorrt":false
}
```

**Reply by painebenjamin** [reply]
That explains some of it - it's my bad for letting you select the `inpainting` checkpoint at all. That's at least part of the problem - you're trying to use an `inpainting` checkpoint for non-inpainting. Switch that model to the base SD model (or don't use a model at all - if there is nothing in the model picker, it defaults to SD v1.5 non-inpainting).

When inpainting is needed, enfugue will automatically switch to the inpainting checkpoint. If there is no inpainting checkpoint, like if you just downloaded a regular one (as is usually the case,) it creates one by merging the base inpainting checkpoint at run-time.

**Reply by ryan13mt** [reply]
To be honest i tried both models that where already installed. The below got the same cuda error, maybe there is no limit being set on how much memory to allocate? I get the same error on auto1111 when i try to generate anything larger than 1536x1536 for example.

```
{
   "model":"C:\\Users\\User\\.cache\\enfugue\\checkpoint\\v1-5-pruned.ckpt",
   "lora":[
      
   ],
   "inversion":[
      
   ],
   "width":512,
   "height":512,
   "size":512,
   "seed":null,
   "prompt":"orc character",
   "negative_prompt":"",
   "image":null,
   "image_callback_steps":10,
   "nodes":[
      {
         "width":512,
         "height":512,
         "prompt":"orc character",
         "negative_prompt":"",
         "controlnet":null,
         "conditioning_scale":1,
         "strength":0.8,
         "num_inference_steps":50,
         "guidance_scale":7.5,
         "remove_background":false,
         "process_control_image":true,
         "scale_to_model_size":true,
         "image":null,
         "mask":null,
         "control_image":null,
         "children":[
            
         ],
         "bounds":[
            [
               0,
               0
            ],
            [
               512,
               512
            ]
         ]
      }
   ],
   "samples":1,
   "upscale":{
      "method":[
         "esrgan"
      ],
      "amount":1,
      "iterative":false,
      "diffusion":false
   },
   "chunking_size":64,
   "chunking_blur":64,
   "build_tensorrt":false
}
```

**Reply by painebenjamin** [reply]
Okay - thanks a lot for all of the information. I'm going to have to do some more digging, I'll let you know what I come up with.

Thanks again!

**Reply by ryan13mt** [reply]
I'll keep a look at the repo for more updates in the future. Will leave this as unanswered if you dont mind maybe other people encounter it and help in finding a solution. Should i open an issue instead?

**Reply by ryan13mt** [reply, marked as answer]
Opened https://github.com/painebenjamin/app.enfugue.ai/issues/27
